docs/custom_math/syntax.md

```diff
@@ -0,0 +1,218 @@
+
+Combining `foreach` and `where` will create an n-dimensional boolean array.
+Wherever index items in this array are _True_, your component `expression(s)` will be applied.
+
+## expression strings
+
+As with where strings, expression strings are a series of math terms combined with operators.
+The terms can be input parameters, decision variables, global expressions, or numeric values that you define on-the-fly.
+
+If you are defining a `global expression` or `objective`, then the available expression string operators are: `+`, `-`, `*`, `/`, and `**` ("to the power of").
+These expressions are applied using standard operator precedence (BODMAS/PEMDAS, see [this wiki](https://en.wikipedia.org/wiki/Order_of_operations) for more info).
+
+If you are defining a `constraint`, then you also need to define a comparison operator: `<=`, `>=`, or `==`.
+
+??? examples
+
+    - If you want to limit all technology outflow to be less than 200 units: `flow_out <= 200`.
+    - If you want to create a global expression which is the storage level minus a parameter defining a minimum allowed storage level: `storage - storage_cap * min_storage_level`.
+    - If you want to set the outflow of a specific technology `my_tech` to equal all outflows of a specific carrier `my_carrier` at each node: `flow_out[techs=my_tech] == sum(flow_out[carriers=my_carrier], over=techs)`.
+    - If you want inflows at a node `my_node` to be at least as much as the inflows in the previous timestep: `flow_in[nodes=my_node] >= roll(flow_in[nodes=my_node], timesteps=1)`.
+
+### Slicing data
+
+You do not need to define the sets of math components in expressions, unless you are actively "slicing" them.
+Behind the scenes, we will make sure that every relevant element of the defined `foreach` sets are matched together when applying the expression (we [merge the underlying xarray DataArrays](https://docs.xarray.dev/en/stable/user-guide/combining.html)).
+Slicing math components involves appending the component with square brackets that contain the slices, e.g. `flow_out[carriers=electricity, nodes=[A, B]]` will slice the `flow_out` decision variable to focus on `electricity` in its `carriers` dimension and only has two nodes (`A` and `B`) on its `nodes` dimension.
+To find out what dimensions you can slice a component on, see your input data (`model.inputs`) for parameters and the definition for decision variables in your loaded math dictionary (`model.math.variables`).
+
+### Helper functions
+
+As with [`where` strings][where-strings], there are many [helper function](https://calliope.readthedocs.io/en/latest/api_reference/#calliopebackendhelper_functions-math-formulation-helper-functions) available to use in expressions, to allow for more complex operations to be undertaken.
```

Comment:
Might be helpful to have a manually crafted table of these in addition to the link to the full API docs?

```suggestion
For [`where` strings][where-strings] and [`expression` strings][expression-strings], there are many [helper functions](../reference/api/helper_functions.md) available to use, to allow for more complex operations to be undertaken.
```